pkg/store/cache/inmemory.go

```diff
@@ -360,17 +367,25 @@ func (c *InMemoryIndexCache) FetchMultiSeries(ctx context.Context, blockID ulid.
 	hits = map[storage.SeriesRef][]byte{}
 
 	blockIDKey := blockID.String()
+	requests := 0
+	hit := 0
 	for _, id := range ids {
 		if ctx.Err() != nil {
+			c.commonMetrics.requestTotal.WithLabelValues(cacheTypeSeries, tenant).Add(float64(requests))
+			c.commonMetrics.hitsTotal.WithLabelValues(cacheTypeSeries, tenant).Add(float64(hit))
 			return hits, misses
 		}
-		if b, ok := c.get(cacheTypeSeries, cacheKey{blockIDKey, cacheKeySeries(id), ""}, tenant); ok {
+		requests++
+		if b, ok := c.get(cacheKey{blockIDKey, cacheKeySeries(id), ""}); ok {
+			hit++
 			hits[id] = b
 			continue
 		}
 
 		misses = append(misses, id)
 	}
+	c.commonMetrics.requestTotal.WithLabelValues(cacheTypeSeries, tenant).Add(float64(requests))
```

Comment:
Does it make sense to call `WithLabelValues` at the start and store it into a variable that we can use everywhere?

Reply:
Ah I see what you mean. You mean we call `WithLabelValues` once, store it as a var and then we can call its method multiple times.

I am okay with it, but I don't see much difference between it and my current approach. 
They both call `WithLabelValues` once, but one approach calls `Inc` multiple times, another approach calls `Add` once.
Performance sounds similar as main bottleneck is `WithLabelValues`